tests/cost_tests.cpp

```diff
@@ -27,4 +29,56 @@ TEST_CASE("Linear Cost Test", "[cost]") {
     torc::LinearCost cost2 = torc::LinearCost<float>(q2, name2);
     REQUIRE(cost2.Evaluate(x3) == 26);
     REQUIRE(cost2.Gradient() == q2);
+};
+
+TEST_CASE("Quadratic Cost Test", "[cost]") {
```

Comment:
Can we change the hard-coded test out for something where we randomly generate the matrix, then check all of these values? Then we can run that in a loop say 50 or 100 times to check a bunch of random matrices. Although it's probably not critical here, I think this is gonna help us catch bugs down the line.

Now, I'm not a test writing expert, so idk if it provides us value to keep the hardcoded ones too - probably not though. What do you think?